packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -468,414 +425,314 @@ class _MinuteControl extends StatelessWidget {
-  _TimePickerFragmentContext context,
-  Orientation orientation,
-) {
-  // Creates an hour fragment.
-  _TimePickerHeaderFragment hour() {
-    return _TimePickerHeaderFragment(
-      layoutId: _TimePickerHeaderId.hour,
-      widget: _HourControl(fragmentContext: context),
-    );
-  }
 
-  // Creates a minute fragment.
-  _TimePickerHeaderFragment minute() {
-    return _TimePickerHeaderFragment(
-      layoutId: _TimePickerHeaderId.minute,
-      widget: _MinuteControl(fragmentContext: context),
-    );
-  }
+/// Displays the am/pm fragment and provides controls for switching between am
+/// and pm.
+class _DayPeriodControl extends StatelessWidget {
+  const _DayPeriodControl({
+    @required this.selectedTime,
+    @required this.onChanged,
+    @required this.orientation,
+  });
 
-  // Creates a string fragment.
-  _TimePickerHeaderFragment string(_TimePickerHeaderId layoutId, String value) {
-    return _TimePickerHeaderFragment(
-      layoutId: layoutId,
-      widget: _StringFragment(
-        fragmentContext: context,
-        value: value,
-      ),
-    );
-  }
+  final TimeOfDay selectedTime;
+  final Orientation orientation;
+  final ValueChanged<TimeOfDay> onChanged;
 
-  // Creates an am/pm fragment.
-  _TimePickerHeaderFragment dayPeriod() {
-    return _TimePickerHeaderFragment(
-      layoutId: _TimePickerHeaderId.period,
-      widget: _DayPeriodControl(fragmentContext: context, orientation: orientation),
-    );
+  void _togglePeriod() {
+    final int newHour = (selectedTime.hour + TimeOfDay.hoursPerPeriod) % TimeOfDay.hoursPerDay;
+    final TimeOfDay newTime = selectedTime.replacing(hour: newHour);
+    onChanged(newTime);
   }
 
-  // Convenience function for creating a time header format with up to two pieces.
-  _TimePickerHeaderFormat format(
-    _TimePickerHeaderPiece piece1, [
-    _TimePickerHeaderPiece piece2,
-  ]) {
-    final List<_TimePickerHeaderPiece> pieces = <_TimePickerHeaderPiece>[];
-    switch (context.textDirection) {
-      case TextDirection.ltr:
-        pieces.add(piece1);
-        if (piece2 != null)
-          pieces.add(piece2);
+  void _setAm(BuildContext context) {
+    if (selectedTime.period == DayPeriod.am) {
+      return;
+    }
+    switch (Theme.of(context).platform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.linux:
+      case TargetPlatform.windows:
+        _announceToAccessibility(context, MaterialLocalizations.of(context).anteMeridiemAbbreviation);
         break;
-      case TextDirection.rtl:
-        if (piece2 != null)
-          pieces.add(piece2);
-        pieces.add(piece1);
+      case TargetPlatform.iOS:
+      case TargetPlatform.macOS:
         break;
     }
-    int centerpieceIndex;
-    for (int i = 0; i < pieces.length; i += 1) {
-      if (pieces[i].pivotIndex >= 0) {
-        centerpieceIndex = i;
-      }
+    _togglePeriod();
+  }
+
+  void _setPm(BuildContext context) {
```

Comment:
It doesn't look like this method does anything

Reply:
I think that may just be the github diff tool, it does do something.

```
  void _setPm(BuildContext context) {
    if (selectedTime.period == DayPeriod.pm) {
      return;
    }
    switch (Theme.of(context).platform) {
      case TargetPlatform.android:
      case TargetPlatform.fuchsia:
      case TargetPlatform.linux:
      case TargetPlatform.windows:
        _announceToAccessibility(context, MaterialLocalizations.of(context).postMeridiemAbbreviation);
        break;
      case TargetPlatform.iOS:
      case TargetPlatform.macOS:
        break;
    }
    _togglePeriod();
  }
```